django/db/models/window.py

```diff
@@ -0,0 +1,189 @@
+from django.db.models import Expression
+from django.db.models.expressions import ExpressionList, OrderBy
+from django.utils.functional import cached_property
+
+__all__ = ['RowRange', 'ValueRange', 'Window']
+
+
+class Window(Expression):
+    template = '%(expression)s OVER (%(window)s)'
+    filterable = False
+
+    def __init__(self, expression, partition_by=None, order_by=None, frame=None, output_field=None):
+        self.partition_by = partition_by
+        self.order_by = order_by
+        self.frame = frame
+
+        if not getattr(expression, 'window_compatible', False):
+            raise ValueError('expression must be compatible in an over-clause')
+
+        if self.partition_by is not None:
+            if isinstance(self.partition_by, (tuple, list)):
+                self.partition_by = ExpressionList(*self.partition_by)
+            else:
+                self.partition_by = ExpressionList(self.partition_by)
+
+        if self.order_by is not None:
+            if not isinstance(self.order_by, (OrderBy, Expression, list)):
+                raise ValueError('order_by must either be of type OrderBy or an Expression')
```

Comment:
Error message doesn't match the code, and we're ignoring the possibility of tuples. Not to mention that OrderBy *is* an Expression.

What do we want to happen here? If it's an expression, do we need to order it (by calling asc/desc)? Or do we just want to push any expression into an ExpressionList?